src/main/java/HttpVerifier.java

```diff
@@ -1,12 +1,16 @@
 package com.td.HttpServer;
 
-public class HttpVerifier implements Validator {
+public class HttpVerifier implements IValidator {
 
-  public boolean isValid(HttpProtocal request) {
-
-    return isMethodValid(request.method()) &&
-           isPathValid(request.path()) &&
-           isVersionValid(request.version());
+  public boolean isValid(String[] request) {
+    try {
+      return isMethodValid(request[0]) &&
+        isPathValid(request[1]) &&
+        isVersionValid(request[2]);
+    }
+    catch (IndexOutOfBoundsException e) {
```

Comment:
While this works, it's not ideal to catch an `IndexOutOfBoundsException` if it can easily be checked for.

This also leads to a potentially misleading bug - imagine if your `isMethodValid` tries to incorrectly access some array that it creates throws an IndexOutOfBoundsException and doesn't catch it? Not a fun time debugging that.